app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java

```diff
@@ -667,19 +667,21 @@ else if (args.contains("FIRE_PULSES")) {
         }
     }
 
-    public Map<String, double[]> captureFullSpeed(String channel, int samples, double timeGap, List<String> args, Integer interval) {
+    public HashMap<String, double[]> captureFullSpeed(String channel, int samples, double timeGap, List<String> args, Integer interval) {
         /*
         * Blocking call that fetches oscilloscope traces from a single oscilloscope channel at a maximum speed of 2MSPS
         */
 
```

Comment:
My bad, I initially moved captureFullSpeedInitialize in the onPreExecute method of AsyncTask. Then I moved it in the background thread with captureFullSpeed. Let me do the necessary changes.